### ISSUE TITLE
Import Assets GUI username mismatching

### ISSUE THREAD

**Issue by Jengah** [snipe/snipe-it#3516]
#### Expected Behavior (or desired behavior if a feature request)

Specifying `username` in CSV should assign to user if that user exists in DB.

-----

#### Actual Behavior

CSV uploaded and checked out to wrong user.

-----

#### Please confirm you have done the following before posting your bug report:

- [*] I have enabled debug mode
- [*] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Please provide answers to these questions before posting your bug report:

- Version of Snipe-IT you're running - **3.6.4**
- What OS and web server you're running Snipe-IT on - **Centos7 EC2 Instance**
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) - **Docker**
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error - **Import GUI**
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database - **No**

----
Testing importing with a single Asset.  When Name and Email are not specified, but an existing Username is provided the import completes successfully, but asset is assigned to the wrong user_id.

I can see in DEBUG that the correct username is being pulled in to the select statement (username omitted):

`select * from `users` where (`email` = '' or `username` = '<existing-username>' or `username` = '') and `username` is not null and `users`.`deleted_at` is null limit 1`

However, during the `insert  into assets` statement, the wrong user_id is being set.  

user_id is being set to '3', when the corresponding username in the database is user_id = 21.

I have tried this with multiple different usernames and the user_id is being set to 3 everytime.  Nothing in logs and no browser errors.

**Comment by snipe** [comment]
Check that you have an entry in the `email domain` settings in Admin > Settings.

<img width="830" alt="screen shot 2017-04-25 at 2 53 54 pm" src="https://cloud.githubusercontent.com/assets/197404/25409515/187ce6e6-29c7-11e7-8dd5-53561137091c.png">

**Comment by Jengah** [comment]
I have verified that it is set at Admin > Settings in the same format as you have pasted.

**Comment by snipe** [comment]
@Jengah I admit I'm a little confused. Thousands of people use the importer every day, and I haven't heard of anything like this happening. I do know that there's an odd bug that shows up if you don't have that email address field completed (even if you don't care about the email address it generates), but the importer flat-out using the wrong user when that field is completed is new to me, so I'm not sure how to troubleshoot. Can you tell us any more about your data (fields used, fields left empty, etc)

**Comment by Jengah** [comment]
@snipe The interesting thing is that the user that I was originally testing with is one of the few users who _does_ have an email address. 

For the user import, I imported firstName, lastName, username, location_id, employee_number, and company_id.

For asset import, I was testing my import using Username, Item Name, Category, Model Name, Model Num, Serial, Asset Tag, Location, and cost.  This is the import structure where I was seeing the wrong user_id being assigned. When I swapped Username for Name is when I was able to successfully get assets imported and assigned to the right user.

**Comment by Bishop200** [comment]
I got the same trouble when I import with username. 

I though it was because my username doesn't fit with the Username Format put in the config. My username are 2 number and first letter of first name last name. 

My user have been imported from AD and most have a email adresse with first name last name without dot.

**Comment by Anonymoos3** [comment]
New to snipe-it (and github so I apologize for the formatting) and I appear to be having the exact same issue. 


I created some toner cartridges as consumables and discovered you can only checkout consumables to users, whereas I wanted to checkout to an asset (printer).

Fine, create printer as a user:

First Name = Printer
Last Name = IT
Username = Printer_IT

I still want to have the Printer as an asset and checkout to newly created printer user. Create CSV file with the following headers:

Asset Tag | Serial Number | Notes | Model Name | Category | Status	    | Username | Manufacturer

Printer_IT | xxxx                 | xxxxx  | 2155cdn       | Printer     | Deployed | Printer_IT  | Dell
 
After "successful" import through the webpage, everything looks right but the Checked Out To user is an AD imported user instead of the newly created user I specified.

I am unable to attempt to duplicate the issue on the demo site due to importing being disabled.

**Comment by UniversalSuperBox** [comment]
I've been able to reproduce this on a clean install, and with a couple of very small import csv's. 

[Users.txt](https://github.com/snipe/snipe-it/files/1455276/Users.txt)
[sample-assets.txt](https://github.com/snipe/snipe-it/files/1455275/sample-assets.txt)

This might be an issue related to underscores (or dashes, as I was previously using) in the e-mail field. When importing the Users file, you'll find that the e-mails of the imported users are not populated. 

When following that up by importing the sample-assets file, you'll find that everything is assigned to the first user in the People list without an e-mail. Does Snipe-IT go by e-mail while importing?

**Comment by snipe** [comment]
@UniversalSuperBox Can you be more specific? Which importer are you using? The everything importer, or the one just for users in the People section?

**Comment by UniversalSuperBox** [comment]
@snipe: Interesting! I was not aware of the importer in the People section. I was using the 'everything' importer and setting it to Users. I've now tried to use the People importer, but I'm getting an exception with the following in the logs:

```
[2017-11-15 20:14:05] production.ERROR: Symfony\Component\Debug\Exception\FatalErrorException: Maximum execution time of 30 seconds exceeded in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Hashing/BcryptHasher.php:29
Stack trace:
#0 {main} 
```

Sounds like I have too many users in one file.

**Comment by snipe** [comment]
You can certainly try that. The people importer (versus the universal one) is old and doesn't support that many fields (and will probably be deprecated in the new few versions.)

If you try to upload the MOCK_USERS.csv in the `sample_csvs` directory does that work?

**Comment by UniversalSuperBox** [comment]
Okay, I'll keep using the everything importer then. 

I've uploaded the MOCK_USERS file, but it still doesn't import the e-mail at all. I tried changing one of the e-mails to match the one set in General Settings, but that didn't work either. The logs are silent during an import.

I installed Snipe-IT on Ubuntu 16.04 using the script in the repository. I'm running Apache `2.4.18 (Ubuntu)` and MariaDB `10.1.28-MariaDB-1~xenial`. Snipe-IT version 4.1.3 build 173 (gba38b84).
